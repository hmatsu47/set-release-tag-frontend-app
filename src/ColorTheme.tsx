import { blue, purple } from "@suid/material/colors";
import { createTheme } from "@suid/material/styles";

export const ColorTheme = createTheme({
  palette: {
    primary: {
      main: localStorage.getItem("primaryColor")
        ? (localStorage.getItem("primaryColor") as string)
        : blue[700],
    },
    secondary: {
      main: localStorage.getItem("secondaryColor")
        ? (localStorage.getItem("secondaryColor") as string)
        : purple[500],
    },
  },
});
