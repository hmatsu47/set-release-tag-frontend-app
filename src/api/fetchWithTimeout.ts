export const fetchWithTimeout = async (
  method: string,
  url: string,
  object?: Object
) => {
  const controller = new AbortController();
  const timeout = setTimeout(() => {
    controller.abort();
  }, 10000);

  try {
    const request = object
      ? {
          // object の指定があれば body に JSON 形式でセットしてリクエスト
          method: method,
          headers: {
            Accept: "application/json",
            "Content-Type": "application/json;charset=utf8",
          },
          body: JSON.stringify(object),
          signal: controller.signal,
        }
      : {
          // なければ body なしでリクエスト
          method: method,
          signal: controller.signal,
        };
    const response = await fetch(url, request);
    // このアプリケーションでは Error を throw しない
    // if (!response.ok) {
    //   throw new Error(response.statusText);
    // }
    return response;
  } finally {
    clearTimeout(timeout);
  }
};
