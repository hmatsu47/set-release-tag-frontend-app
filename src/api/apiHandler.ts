import { fetchWithTimeout } from "./fetchWithTimeout";

// ベース URI
export const baseUri = "/api";

// API からデータ取得（GET）
export const getApiData = async (url: string) => {
  return await (await fetchWithTimeout("GET", url)).json();
};
// API からデータ取得（POST）
export const postApiData = async (url: string, object: Object) => {
  return await (await fetchWithTimeout("POST", url, object)).json();
};
