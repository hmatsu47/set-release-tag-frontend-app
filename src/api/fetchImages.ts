import { ErrorResponse, ImageItem } from "../type";
import { baseUri, getApiData } from "./apiHandler";
import {
  setImages,
  setMessage,
  setMessageSeverity,
  setSelectedTag,
} from "../signal";

export const fetchImages = async () => {
  const load = async (): Promise<void> => {
    const data: ImageItem[] | ErrorResponse = await getApiData(
      `${baseUri}/images`
    );
    if (
      typeof data === "object" &&
      data !== null &&
      typeof (data as ErrorResponse).message === "string"
    ) {
      // 戻り値がエラーメッセージの場合
      setImages(undefined);
      setMessage((data as ErrorResponse).message);
      setMessageSeverity("error");
      return;
    }
    setImages(data as ImageItem[]);
    setSelectedTag(undefined);
    setMessage(undefined);
  };
  void load();
};
