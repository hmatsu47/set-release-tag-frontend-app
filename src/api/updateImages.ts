import { ErrorResponse, ImageItem, ImageTag } from "../type";
import { baseUri, postApiData } from "./apiHandler";
import {
  selectedTag,
  setImages,
  setMessage,
  setMessageSeverity,
  setSelectedTag,
} from "../signal";

export const updateImages = async () => {
  if (!selectedTag()) {
    return;
  }
  const load = async (): Promise<void> => {
    const data: ImageItem[] | ErrorResponse = await postApiData(
      `${baseUri}/images`,
      {
        tag: selectedTag(),
      } as ImageTag
    );
    if (
      typeof data === "object" &&
      data !== null &&
      typeof (data as ErrorResponse).message === "string"
    ) {
      // 戻り値がエラーメッセージの場合
      setImages(undefined);
      setMessage((data as ErrorResponse).message);
      setMessageSeverity("error");
      return;
    }
    setImages(data as ImageItem[]);
    setSelectedTag(undefined);
    setMessage(undefined);
  };
  void load();
};
