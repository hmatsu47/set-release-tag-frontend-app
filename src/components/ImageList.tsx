import { For, Show, onMount } from "solid-js";
import Box from "@suid/material/Box";
import Button from "@suid/material/Button";
import Paper from "@suid/material/Paper";
import Table from "@suid/material/Table";
import TableBody from "@suid/material/TableBody";
import TableCell from "@suid/material/TableCell";
import TableContainer from "@suid/material/TableContainer";
import TableHead from "@suid/material/TableHead";
import TableRow from "@suid/material/TableRow";
import Typography from "@suid/material/Typography";
import { fetchImages } from "../api/fetchImages";
import { images, setIsOpenedConfirm, setSelectedTag } from "../signal";
import { formatDateTimeDisplay } from "../formatDateTime";

export const ImageList = () => {
  onMount(async () => {
    if (images() == undefined) {
      await fetchImages();
    }
  });

  return (
    <Show when={images() !== undefined} fallback={<></>}>
      <Box
        sx={{
          width: "100%",
          minWidth: "1024px",
          display: "flex",
        }}
        aria-live="polite"
      >
        <Show
          when={images() !== null}
          fallback={
            <Typography variant="h6">
              リポジトリにイメージがありません
            </Typography>
          }
        >
          <TableContainer component={Paper}>
            <Table aria-label="simple table">
              <TableHead>
                <TableRow>
                  <TableCell>選択</TableCell>
                  <TableCell>イメージタグ</TableCell>
                  <TableCell>プッシュ日時</TableCell>
                  <TableCell>
                    サイズ
                    <br />
                    （MB）
                  </TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                <For each={images()} fallback={<></>}>
                  {(imageItem) => (
                    <TableRow>
                      <TableCell component="th" scope="row">
                        <Button
                          variant={
                            imageItem.tags.filter((n) => n === "release")
                              .length > 0
                              ? "contained"
                              : "outlined"
                          }
                          size="small"
                          color={
                            imageItem.tags.filter((n) => n === "release")
                              .length > 0
                              ? "primary"
                              : "inherit"
                          }
                          onClick={async (e) => {
                            if (
                              imageItem.tags.filter((n) => n === "release")
                                .length == 0
                            ) {
                              setSelectedTag(imageItem.tags[0]);
                              setIsOpenedConfirm(true);
                            }
                          }}
                          sx={{ textTransform: "none" }}
                          title={`${imageItem.tags.join(", ")} を選択`}
                        >
                          選択
                        </Button>
                      </TableCell>
                      <TableCell>{imageItem.tags.join(", ")}</TableCell>
                      <TableCell>
                        {formatDateTimeDisplay(new Date(imageItem.pushed_at))}
                      </TableCell>
                      <TableCell align="right">
                        {(
                          Math.round(
                            (imageItem.size / (1000 * 1000)) * Math.pow(10, 2)
                          ) / Math.pow(10, 2)
                        ).toLocaleString()}
                      </TableCell>
                    </TableRow>
                  )}
                </For>
              </TableBody>
            </Table>
          </TableContainer>
        </Show>
      </Box>
    </Show>
  );
};
