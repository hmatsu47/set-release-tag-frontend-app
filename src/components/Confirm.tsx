import { Show } from "solid-js";
import { useTheme } from "@suid/material";
import Backdrop from "@suid/material/Backdrop";
import Box from "@suid/material/Box";
import Button from "@suid/material/Button";
import Stack from "@suid/material/Stack";
import Typography from "@suid/material/Typography";
import { updateImages } from "../api/updateImages";
import { isOpenedConfirm, setIsOpenedConfirm } from "../signal";

export const Confirm = () => {
  const theme = useTheme();

  return (
    <Show when={isOpenedConfirm() === true} fallback={<></>}>
      <Backdrop
        sx={{ color: "#fff", zIndex: theme.zIndex.drawer + 1 }}
        open={isOpenedConfirm()}
      ></Backdrop>
      <Box
        sx={{
          position: "absolute",
          top: "25%",
          left: "50%",
          transform: "translate(-50%, -50%)",
          width: 400,
          bgcolor: theme.palette.background.paper,
          border: "2px solid #000",
          boxShadow: "24px",
          p: 4,
          zIndex: theme.zIndex.drawer + 1,
        }}
      >
        <Typography variant="subtitle1">
          本番リリース時に使用するイメージを選択します。
        </Typography>
        <Typography variant="subtitle1">よろしいですか？</Typography>
        <Box sx={{ paddingTop: "20px" }}></Box>
        <Stack spacing={4} direction="row" sx={{ justifyContent: "center" }}>
          <Button
            variant="contained"
            size="small"
            color="primary"
            onClick={async (e) => {
              setIsOpenedConfirm(false);
              await updateImages();
            }}
            title="はい"
          >
            はい
          </Button>
          <Button
            variant="outlined"
            size="small"
            color="inherit"
            onClick={() => {
              setIsOpenedConfirm(false);
            }}
            title="いいえ"
          >
            いいえ
          </Button>
        </Stack>
      </Box>
    </Show>
  );
};
