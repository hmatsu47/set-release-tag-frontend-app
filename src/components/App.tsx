import { Component } from "solid-js";
import Box from "@suid/material/Box";
import Stack from "@suid/material/Stack";
import { ThemeProvider } from "@suid/material";
import { ColorTheme } from "../ColorTheme";
import { ImageList } from "./ImageList";
import { Message } from "./Message";
import { TitleBar } from "./TitleBar";
import { Confirm } from "./Confirm";

export const App: Component = () => {
  return (
    <ThemeProvider theme={ColorTheme}>
      <TitleBar />
      <Box
        sx={{
          padding: "10px 10px 0 10px",
          width: "100%",
          minWidth: "1024px",
          display: "flex",
          // justifyContent: "center",
        }}
        aria-live="polite"
      >
        <Stack spacing={2} direction="column">
          <Message />
          <ImageList />
          <Confirm />
        </Stack>
      </Box>
    </ThemeProvider>
  );
};
