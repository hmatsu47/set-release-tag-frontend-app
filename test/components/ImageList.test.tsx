// @vitest-environment jsdom

import { describe, expect, test, vi } from "vitest";
import "vi-fetch/setup";
import { mockFetch, mockGet } from "vi-fetch";
import { fireEvent, render } from "@solidjs/testing-library";
import { formatSnapshot } from "../common/formatSnapshot";
import { ImageList } from "../../src/components/ImageList";
import { baseUri } from "../../src/api/apiHandler";
import {
  isOpenedConfirm,
  selectedTag,
  setImages,
  setIsOpenedConfirm,
  setSelectedTag,
} from "../../src/signal";
import { ImageItem } from "../../src/type";

describe("<ImageList />", () => {
  beforeEach(() => {
    mockFetch.clearAll();
  });
  afterEach(() => {
    vi.clearAllMocks();
  });
  // 一覧表示されるケース
  const imageCheckList = [
    {
      title: "Confirm が表示されない（前回選択と同じボタンをクリック）",
      clickTag: "hoge, release",
      attachTag: "release",
      expectedTag: "hoge",
      expectedOpenConfirm: false,
    },
    {
      title: "Confirm が表示される（前回選択と違うボタンをクリック）",
      clickTag: "hoge, release",
      attachTag: "latest",
      expectedTag: "hoge",
      expectedOpenConfirm: true,
    },
  ];
  imageCheckList.forEach((testCase) => {
    test(testCase.title, async () => {
      localStorage.setItem("attachTagName", testCase.attachTag);
      setIsOpenedConfirm(false);
      // まずは一覧非表示の状態
      setSelectedTag(undefined);
      setImages(null);
      const mock = mockGet(`${baseUri}/images`).willResolve();
      const { container, findByText, findByTitle, unmount } = render(() => (
        <ImageList />
      ));
      expect(mock).toHaveFetched();
      // API 呼び出しチェックが終わったら必要項目をセットして確認
      setSelectedTag("hoge");
      setImages([
        {
          digest:
            "sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d",
          pushed_at: new Date(2022, 8, 12, 14, 45),
          repository_name: "hoge",
          size: 10000,
          tags: ["hoge", "release"],
        } as ImageItem,
        {
          digest:
            "sha256:4c716d4cf211c7b7d2f3233c941771ad0507ea5bacf93b492766aa41ae9f720d",
          pushed_at: new Date(2022, 8, 12, 14, 45),
          repository_name: "hoge",
          size: 2000000,
          tags: ["latest"],
        } as ImageItem,
      ]);
      const expectedListedTag = (await findByText(
        testCase.clickTag
      )) as HTMLElement;
      expect(expectedListedTag).toHaveTextContent(testCase.clickTag);
      // css の名前が動的に変わるので固定値に置換
      const htmlBefore = formatSnapshot(container.innerHTML);
      expect(htmlBefore).toMatchSnapshot();
      // ボタンクリック
      const button = (await findByTitle(
        `${testCase.clickTag} を選択`
      )) as HTMLInputElement;
      expect(button).toHaveTextContent("選択");
      fireEvent.click(button);
      expect(isOpenedConfirm()).toBe(testCase.expectedOpenConfirm);
      expect(selectedTag()).toBe(testCase.expectedTag);
      // イメージタグ指定後のスナップショット
      const htmlAfterTagSelect = formatSnapshot(container.innerHTML);
      expect(htmlAfterTagSelect).toMatchSnapshot();
      unmount();
    });
  });
  // 一覧表示されないケース（イメージなし）
  test("一覧未表示（イメージなし）", async () => {
    // すべて無指定
    setIsOpenedConfirm(false);
    setSelectedTag(undefined);
    setImages(null);
    const mock = mockGet(`${baseUri}/images`).willResolve();
    const { container, findByText, unmount } = render(() => <ImageList />);
    expect(mock).toHaveFetched();
    const expected = (await findByText(
      "リポジトリにイメージがありません"
    )) as HTMLElement;
    expect(expected).toHaveTextContent("リポジトリにイメージがありません");
    // css の名前が動的に変わるので固定値に置換
    const html = formatSnapshot(container.innerHTML);
    expect(html).toMatchSnapshot();
    unmount();
  });
});
