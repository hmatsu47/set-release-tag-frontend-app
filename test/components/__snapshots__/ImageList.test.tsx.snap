// Vitest Snapshot v1

exports[`<ImageList /> > Confirm が表示されない（前回選択と同じボタンをクリック） 1`] = `"<div aria-live=\\"polite\\" class=\\"MuiBox-root css-xxxxxxxx\\"><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table aria-label=\\"simple table\\" class=\\"MuiTable-root css-xxxxxxxx\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">選択</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">イメージタグ（注：タグのないイメージは一覧から除外）</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">プッシュ日時</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">サイズ（MB）</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th scope=\\"row\\" class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"hoge, release を選択\\" tabindex=\\"0\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">hoge, release</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">0.01</td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th scope=\\"row\\" class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"latest を選択\\" tabindex=\\"0\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">latest</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2</td></tr></tbody></table></div></div>"`;

exports[`<ImageList /> > Confirm が表示されない（前回選択と同じボタンをクリック） 2`] = `"<div aria-live=\\"polite\\" class=\\"MuiBox-root css-xxxxxxxx\\"><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table aria-label=\\"simple table\\" class=\\"MuiTable-root css-xxxxxxxx\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">選択</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">イメージタグ（注：タグのないイメージは一覧から除外）</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">プッシュ日時</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">サイズ（MB）</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th scope=\\"row\\" class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"hoge, release を選択\\" tabindex=\\"0\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">hoge, release</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">0.01</td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th scope=\\"row\\" class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"latest を選択\\" tabindex=\\"0\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">latest</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2</td></tr></tbody></table></div></div>"`;

exports[`<ImageList /> > Confirm が表示される（前回選択と違うボタンをクリック） 1`] = `"<div aria-live=\\"polite\\" class=\\"MuiBox-root css-xxxxxxxx\\"><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table aria-label=\\"simple table\\" class=\\"MuiTable-root css-xxxxxxxx\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">選択</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">イメージタグ（注：タグのないイメージは一覧から除外）</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">プッシュ日時</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">サイズ（MB）</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th scope=\\"row\\" class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"hoge, release を選択\\" tabindex=\\"0\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">hoge, release</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">0.01</td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th scope=\\"row\\" class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"latest を選択\\" tabindex=\\"0\\">選択<span class=\\"MuiTouchRipple-root css-xxxxxxxx\\"></span></button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">latest</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2</td></tr></tbody></table></div></div>"`;

exports[`<ImageList /> > Confirm が表示される（前回選択と違うボタンをクリック） 2`] = `"<div aria-live=\\"polite\\" class=\\"MuiBox-root css-xxxxxxxx\\"><div class=\\"MuiTableContainer-root css-xxxxxxxx\\"><table aria-label=\\"simple table\\" class=\\"MuiTable-root css-xxxxxxxx\\"><thead class=\\"MuiTableHead-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">選択</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">イメージタグ（注：タグのないイメージは一覧から除外）</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">プッシュ日時</td><td scope=\\"col\\" class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">サイズ（MB）</td></tr></thead><tbody class=\\"MuiTableBody-root css-xxxxxxxx\\"><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th scope=\\"row\\" class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"hoge, release を選択\\" tabindex=\\"-1\\" disabled=\\"\\">選択</button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">hoge, release</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">0.01</td></tr><tr class=\\"MuiTableRow-root css-xxxxxxxx\\"><th scope=\\"row\\" class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\"><button type=\\"button\\" class=\\"MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButton-root MuiButtonBase-root css-xxxxxxxx\\" title=\\"latest を選択\\" tabindex=\\"-1\\" disabled=\\"\\">選択</button></th><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">latest</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2022-09-12 14:45</td><td class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium MuiTableCell-root css-xxxxxxxx\\">2</td></tr></tbody></table></div></div>"`;

exports[`<ImageList /> > 一覧未表示（イメージなし） 1`] = `"<div aria-live=\\"polite\\" class=\\"MuiBox-root css-xxxxxxxx\\"><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">リポジトリにイメージがありません</h6></div>"`;
